### DISCUSSION TITLE
HTTP & JSON Konfiguration

### DISCUSSION THREAD

**Discussion by iukbox** [hoylabs/OpenDTU-OnBattery#191] [Q&A] [answered]
Hallo

ich habe die URL für den Webrequest vom "Volkszähler" mit Tasmota bei den Stromzähler Einstellungen angegeben:
http://xxx.xxx.xxx.xxx/cm?cmnd=status%208
Die Antwort ist:
{"StatusSNS":{"Time":"2023-04-26T20:57:24","MT681":{"Total_in":1350.784,"Power_cur":439,"Total_out":93.151}}}
Was muss denn nun in den JSON Pfad? **Power_cur** führt zur folgenden Meldung: Error: Could not find value for JSON path!

**Comment by madmartin** [comment, marked as answer]
Es müsste sein

`StatusSNS/MT681/Power_cur`

Da sollte definitiv ein Beispiel in die zukünftige Doku....

**Reply by iukbox** [reply]
Danke, so gehts.

**Reply by KJ-202209** [reply]
Hallo ich habe auch so ein Problem. Leider funktioniert es nicht. Beim Test bekomme ich keine Antwort. Ich habe folgendes eingetragen:
in DTU:
ERHALTEN
http://192.168.0.92/cm?cmnd=status%208
Authentifizierungsverfahren

Ohne
HTTP-Header - Name
HTTP Header - Wert
Zeitüberschreitung
1000
Millisekunden
JSON-Pfad
StatusSNS/SML/curr_w
Einheit

B
Vorzeichen UM kehren

**Mein SKRIPT:**
>D
>B
->sensor53 r
>M 1
+1,3,s,16,9600,SML
1,77070100600100ff@#,Server ID,,server_id,0
1,77070100020800ff@1000,Export (Total),kWh,export_total_kwh,4
1,77070100010800ff@1000,Consumption (Total),kWh,total_kwh,4
1,77070100100700ff@1,Consumption (Current),W,curr_w,0
1,77070100200700ff@1,Voltage L1,V,volt_p1,1
1,77070100340700ff@1,Voltage L2,V,volt_p2,1
1,77070100480700ff@1,Voltage L3,V,volt_p3,1
1,770701001f0700ff@1,Amperage L1,A,amp_p1,1
1,77070100330700ff@1,Amperage L2,A,amp_p2,1
1,77070100470700ff@1,Amperage L3,A,amp_p3,1
1,77070100510701ff@1,Phase angle U-L2/U-L1,deg,phase_angle_l2_l1,1
1,77070100510702ff@1,Phase angle U-L3/U-L1,deg,phase_angle_l3_l1,1
1,77070100510704ff@1,Phase angle I-L1/U-L1,deg,phase_angle_p1,1
1,7707010051070fff@1,Phase angle I-L2/U-L2,deg,phase_angle_p2,1
1,7707010051071aff@1,Phase angle I-L3/U-L3,deg,phase_angle_p3,1
1,770701000e0700ff@1,Frequency,Hz,freq,0
#

**Reply by Manos1966** [reply]
@KJ-202209 
OK, ich sehe....
ich habe das Wiki geupdated, jetzt steht:

The standard command http://192.168.xxx.xxx/cm?cmnd=status%208 (where 192.168.xxx.xxx is the IP Adress of your Tasmota) which, **when entered in your Web Browser,** will return the values that are availabl

**Comment by peff74** [comment]
Hi,

kann mir jemand dann auch bitte helfen, wie ich hier die Daten rausbekomme?

`["11038.330","  149","55 days, 4 hours, 37 minutes, 35 seconds","37.09 KB","26.61 KB","21","32","-75","DC:39:6F:DB:4C:BD","ESP8266_SML_v17","Nov 23 2022","14:36:06","Software/System restart"]`

Es handelt sich dabei um den 2. Wert.


Danke :-)

**Reply by iukbox** [reply]
Das sieht aber nicht nach einem JSON String aus. Woher kommen denn die Daten?

**Reply by peff74** [reply]
die Lösung ist [1] !

Da bastelt man 45min rum, liest alles quer, schreibt schon verzweifelt um Hilfe....
...und dann wird's einem Klar^^

**Reply by peff74** [reply]
doch is einer, den nutzte ich in meinem ESP am Stromzähler um die Website vom ESP upzudaten.
Missbrauche Ihn jetzt quasi hierfür

**Reply by madmartin** [reply]
Ist halt nur ein JSON Array, geht auch ganz ohne `{}`

**Reply by iukbox** [reply]
Ah, vielleicht sollte ich mich doch mal eingehender mit JSON beschäftigen. :)

**Comment by codesache** [comment]
Hallo zusammen,
bei mir steht an der Stelle kein Wert nach dem ON für den Type.

{"StatusSNS":{"Time":"2024-02-28T11:10:57","Switch1":"ON","":{"Total":6092.4684,"Supply":327.3241,"Power":347,"Voltage":230.5,"Voltage_L2":229.3,"Voltage_L3":228.9,"Current":0.47,"Current_L2":1.12,"Current_L3":0.94,"Freq":50.0}}}

Was trage ich in Tasmota ein damit der JSON Pfad den Wert bekommt, oder kann ich den ohne dem Wert auslesen ?

StatusSNS/????/Power

**Reply by spcqike** [reply]
kann es sein, dass dein SML meter keinen Namen hat im Tasmota Skript?

**Comment by codesache** [comment]
Ich dachte ich hätte an allen Stellen einen Namen vergeben.

Ist das unter edit script oder Config > Template parameters zu finden.
Ich habe die Einstellungen im Template unter Based on auf Shell 1 das vermute ich mit dem Switch1 in json. Sonst stelle ich das gerne auf was anderes.

**Reply by spcqike** [reply]
das ist im Skript. 

https://tasmota.github.io/docs/Smart-Meter-Interface/#meter-definition

```
+<M>,<rxGPIO>,<type>,<flag>,<parameter>,<jsonPrefix>{,<txGPIO>,<txPeriod>,<cmdTelegram>}

 <jsonPrefix> | Prefix for Web UI and MQTT JSON payload. Up to 7 characters.
```

**Comment by codesache** [comment]
Kannst du mir die Stelle bitte zeigen wo ich den jsonPrefix einfügen soll.
Vielen Dank schon mal! 

>D
>B
->sensor53 r
>M 1
+1,3,s,16,9600,
1,77070100010800ff@1000,Gesamtverbrauch,kWh,Total,4
1,=h<hr/>
;for meters measuring feed-in SGM-Cx-xxx2xx
1,77070100020800ff@1000,Einspeisung,kWh,Supply,4
;next 4 lines for double-tariff meters SGM-Cx-xxxxTx
;1,77070100010801ff@1000,Comsumption_t1,kWh,Total_t1,4
;1,77070100010802ff@1000,Comsumption_t2,kWh,Total_t2,4
;1,77070100020801ff@1000,Supply_t1,kWh,Supply_t1,4
;1,77070100020802ff@1000,Supply_t2,kWh,Supply_t2,4
;all commented lines from here on for 3-phase meter SGM-C4
1,77070100100700ff@1,Netzbezug,W,Power,0
1,=h<hr/>
1,77070100200700ff@1,Volt L1,V,Voltage,1
1,77070100340700ff@1,Volt L2,V,Voltage_L2,1
1,77070100480700ff@1,Volt L3,V,Voltage_L3,1
1,=h<hr/>
1,770701001f0700ff@1,Leistung L1,A,Current,2
1,77070100330700ff@1,Leistung L2,A,Current_L2,2
1,77070100470700ff@1,Leistung L3,A,Current_L3,2
1,=h<hr/>
;1,77070100510701ff@1,Phaseangle L2-L1,deg,phase_angle_L2_L1,0
;1,77070100510702ff@1,Phaseangle L3-L1,deg,phase_angle_L3_L1,0
;1,77070100510704ff@1,Phaseangle I/U L1,deg,phase_angle_L1,1 
;1,7707010051070fff@1,Phaseangle I/U L2,deg,phase_angle_L2,1  
;1,7707010051071aff@1,Phaseangle I/U L3,deg,phase_angle_L3,1 
1,770701000e0700ff@1,Frequency,Hz,Freq,1
;all commented lines from here on just for completeness
;1,7707010000020000@1,Firmware Version,,FwVer,0
;1,77070100605a0201@1,Firmware Checksum,,FwCheck,0
;1,7707010061610000@1,Error Register,,ErrorReg,0
;1,7707010060320101@1,Hardware Version,,HwVer,0
;1,7707010060320104@1,Parameter Version,,ParamVer,0
;1,77070100600100ff@#,Server-ID,,ID,0
;You can find your server-id printed on your meter. If you want you can also convert it to your Identifikationsnummer with some ASCII, DEC and HEX encoding. 0A-01-45-46-52-ST-UV-WX-YZ = 1EFR + string(S) + string(T) + hexToDec(UVWXYZ)
#

**Reply by spcqike** [reply]
```
+1,3,s,16,9600,<prefix>
```

**Comment by codesache** [comment]
@spcqike Danke für deinen Support jetzt bekomme ich die Daten in die DTU

**Comment by codesache** [comment]
@spcqike 
Die Zeile für Gesamtverbrauch schein nach der Erweiterung nicht mehr zu funktionieren und wird auf der Übersicht nicht ausgegeben. Auch in dem cm?cmnd=status%208 ist die nicht mehr enthalten.

1,77070100010800ff@1000,Gesamtverbrauch,kWh,Total,4

Kannst du mir einen Tipp / Hinweis geben was geändert werden muss.

**Comment by codesache** [comment]
Hab's gefixt hinter der Zeile war ein Komma zu viel drinnen !
+1,3,s,16,9600,zaehler

**Comment by 3joern** [comment]
Ahoi, sorry wenn ich auch hier reinkrätsche, das klappt bei mir ledier nicht so wie ich das gerne hätte:

Console:
15:07:13.254 RSL: STATUS8 = {"StatusSNS":{"Time":"2024-03-22T15:07:13","E320":{"Total_out":0.000,"Total_in":12032.827,"Power_in":110.000}}}
15:07:13.539 RSL: STATUS8 = {"StatusSNS":{"Time":"2024-03-22T15:07:13","E320":{"Total_out":0.000,"Total_in":12032.827,"Power_in":110.000}}}
15:07:23.796 RSL: STATUS8 = {"StatusSNS":{"Time":"2024-03-22T15:07:23","E320":{"Total_out":0.000,"Total_in":12032.828,"Power_in":88.000}}}
15:07:24.096 RSL: STATUS8 = {"StatusSNS":{"Time":"2024-03-22T15:07:24","E320":{"Total_out":0.000,"Total_in":12032.828,"Power_in":88.000}}}

Skipt:
>D
>B
=>sensor53 r
>M 1
+1,3,s,20,9600,E320
1,77070100020800ff@1000,Total Delivered,KWh,Total_out,3
1,77070100010800ff@1000,Total Consumed,KWh,Total_in,3
1,77070100100700ff@1,Current power,W,Power_in,3

#

Fehler:
[HttpPowerMeter] Couldn't find a value for phase 0 with Json query "+1,3,s,20,9600,E320"

![Bildschirmfoto 2024-03-22 um 15 06 19](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/6789008d-5a01-4d52-a2e1-143590e742a9)

**Reply by spcqike** [reply]
+1,3,s,20,9600,E320 Ist kein gültiges JSON query. Das ist teil seines tasmota skripts. 

Dein query sollte

StatusSNS/E320/Power_in

sein.

**Reply by 3joern** [reply]
Hui das wars, hab Dank.
Kann man eigentlich auch noch "zusätzlich" den Gesamt Stand anzeigen, also neben StatusSNS/E320/Power_in auch 
StatusSNS/E320/Total_in? wenn ich das in Phase 2 einsetze ist das zwar ok, wird aber auf dem Startscreen nicht angezeigt.

**Reply by spcqike** [reply]
Mein. Du kannst keinen Zählerstand anzeigen lassen. 

Ein Zählerstand in phase 2 würde zu einer falschen Interpretation der Daten führen. Er erwartet ausschließlich aktuelle Leistung.

**Comment by Madworscht** [comment]
Hi,

ich bekomm auch nicht beine Daten aus dem Skipt raus. Ich bekomme folgendes ausgegeben:

`[
    {
        "Uhrzeit": 1746548091,
        "value": -476.4548645019531,
        "Name": "Verbrauch"
    }
]`

Ich bekomme aber leider den nötigen JSON-Pfad für value nicht raus.

Ich hoffe Ihr könnt mir da helfen.

**Reply by schlimmchen** [reply]
Bitte teste `[0]/value`.

**Reply by Madworscht** [reply]
Super hat geklappt. War ich schon nah dran mit [0].value. Aber knapp daneben ist leider auch vorbei. :-)

Danke schön!

**Comment by somermaxer-ai** [comment]
Hi in die Runde, 
ich komme leider mit dem Einlesen der Hichi Lesekopf Daten (Tasmota in Open DTU) auch nicht weiter.
Hier mein script:
>D
>B
=>sensor53 r
;Set teleperiod to 20sec  
tper=10  
>M 1
+1,3,s,0,9600,Power
1,77070100600100ff@#,Zählernummer,,Meter_Number,0
1,77070100010800ff@1000,Verbrauch,kWh,Total_in,4
1,77070100100700ff@1,Leistung,W,Power_curr,0
1,77070100020800ff@1000,Erzeugung,kWh,Total_out,4
#

beim Auslesen über http://192.168.XXX.XX/cm?cmnd=status%25/208 erhalte ich folgende Antwort:
{"Status":{"Module":1,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_FA2FB8","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}

was gehört da jetzt in den JSON Pfad
ich wäre glücklich wenn da jemand helfen könnte

**Reply by Sorbat** [reply]
Hast du schon mal einen Blick ins [Wiki](https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Power-Meter) geworfen?

**Reply by somermaxer-ai** [reply]
@Sorbat,
ja habe ich und habe auch schon mit diversen Eingaben versucht aber leider vergeblich

**Reply by spcqike** [reply]
ist die URL falsch? warum ``status%25/208``? ist es nicht ``status%208``?

**Reply by somermaxer-ai** [reply]
ja ich habe definitiv /cm?cmnd=status%208 aktiv, war wohl copy & paste Fehler hier in den post

**Reply by spcqike** [reply]
prüf das nochmal im Browser. der JSON Response oben passt definitiv nicht zum Status 8. Wenn du dort durch copy-paste auch diesen ``status%25/208`` in der URL hattest, gabs bei Tasmota vielleicht den Fallback auf den normalen ``Status``.

Prüfe es nochmal im Browser, poste den korrekten JSON, der auch die Smartmeter ausgabe enthält, hier.

**Reply by Manos1966** [reply]
Browser Fenster in "Incognito Mode" Macht normalerweise den Trick.

**Reply by somermaxer-ai** [reply]
ich habe es nochmal geprüft mit /cm?cmnd=status%208 - das Ergebnis bleibt gleich, auch im Incognito Modus:

{"Status":{"Module":1,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_FA2FB8","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}

**Reply by spcqike** [reply]
ist leider immer noch nicht die korrekte antwort. irgendwas scheint da nicht zu passen bei dir.

ich habs grad hier nochmal getestet. wenn das ``/cm?cmnd=statusxyz`` irgendwie falsch ist, gibt er nur ``status`` zurück. das bekommst du dort.
``status 8`` sieht anders aus. das brauchst du aber.

im Browser kannst du auch ganz normal ``/cm?cmnd=status 8`` eingeben. mit Leerzeichen statt ``%20`` 

Anhand deines Skriptes sollte der JSON bei dir eigentlich ``StatusSNS/Power/Leistung`` (oder doch ``StatusSNS/Power/Power_curr`` ?) sein. sicher sehen würde man es nur im JSON response.

**Reply by somermaxer-ai** [reply]
hier das Ergebnis mit /cm?cmnd=status 8:

{"StatusSNS":{"Time":"2025-08-15T08:40:17","Power":{"Meter_Number":"0a0149545200034793fc","Total_in":23665.0590,"Power_curr":-437,"Total_out":0.0000}}}

**Reply by somermaxer-ai** [reply]
also jetzt funzt es, in der URL alles belassen mit %208 und im JSON Pfad StatusSNS/Power/Power_curr

Vielen lieben Dank vom älteren Semester und Laien Oli

**Comment by Manos1966** [comment]
zum Verstaendnis:
due hast dieses Script
`D
B
=>sensor53 r
;Set teleperiod to 20sec
tper=10
M 1
+1,3,s,0,9600,Power
1,77070100600100ff@#,Zählernummer,,Meter_Number,0
1,77070100010800ff@1000,Verbrauch,kWh,Total_in,4
1,77070100100700ff@1,Leistung,W,Power_curr,0
1,77070100020800ff@1000,Erzeugung,kWh,Total_out,4
`

in Tasmota eingegeben.

Was zeigt also dein Tasmota fuer Werte, wenn du die TASMOTA Webseite 192.168.xxx.xx aufrufst?

**Reply by somermaxer-ai** [reply]
auf der Tasmota Website sehe ich Zählernummer, Gesamtverbrauch und Leistung (was gerade aus dem Netz entnommen oder eingespeist (mit minus Vorzeichen) wird
